docs/development/extensions-contrib/prometheus.md

```diff
@@ -47,6 +46,8 @@ All the configuration parameters for the Prometheus emitter are under `druid.emi
 | `druid.emitter.prometheus.pushGatewayAddress` | Pushgateway address. Required if using `pushgateway` strategy.                                                                                                                                                                         | no        | none                                 |
 | `druid.emitter.prometheus.flushPeriod`        | Emit metrics to Pushgateway every `flushPeriod` seconds. Required if `pushgateway` strategy is used.                                                                                                                                   | no        | 15                                   |
 | `druid.emitter.prometheus.extraLabels`        | JSON key-value pairs for additional labels on all metrics. Keys (label names) must match the regex `[a-zA-Z_:][a-zA-Z0-9_:]*`. Example: `{"cluster_name": "druid_cluster1", "env": "staging"}`.                                        | no        | none                                 |
+| `druid.emitter.prometheus.deletePushGatewayMetricsOnShutdown ` | Flag to delete metrics from Pushgateway on task shutdown. Works only if `pushgateway` strategy is used.                                                                                                                                                                                                                                                            | no | false |
+| `druid.emitter.prometheus.waitForShutdownDelay`        | Time in milliseconds to wait for peon tasks to delete metrics from the Pushgateway on shutdown (e.g. 60_000). Applicable only when `pushgateway` strategy is used and `pushGatewayDeleteOnShutdown` is set to true. Be aware that there's no guarantee that a peon task will delete metrics from the gateway if the configured delay is more than the [Peon's `druid.indexer.task.gracefulShutdownTimeout`](https://druid.apache.org/docs/latest/configuration/#additional-peon-configuration). | no | none  |
```

Comment:
Also, a guidance on what this value should be set to would be helpful for operators. I think something like 1.2 times the configured Prometheus `scrape_interval` from the push gateway; otherwise, pushed metrics could be lost if they're not scraped promptly from the gateway, similar to the `exporter` model.